src/parser/mod.rs

```diff
@@ -484,8 +488,9 @@ impl<'a> Parser<'a> {
             }
 
             let statement = self.parse_statement()?;
+            // Treat batch delimiter as an end of statement, so no additional statement delimiter expected here
```

Comment:
```suggestion
            // The `GO` statement optionally ends with a semicolon
```
thinking of a this more straightforward description, it would not be obvious to a reader unfamiliar with the statement that 'batch delimiter' is the same thing in this context

Reply:
Done 👍